core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java

```diff
@@ -191,6 +211,151 @@ public void process(WatchedEvent event) {
+     * value of the table-config-version znode will also be changed in addition to the actual table
+     * configuration property (that will no longer be watched). Everytime the data in the
+     * table-config-version znode changes the internal "data version" tracked by Zookeeper will be
+     * automatically incremented. Accumulo does not have to compute what that version is. Zookeeper
+     * does that by itself. The ZooCache object will track this version and use it to maintain the
+     * cache. Many processes will be using it so when the "lastTableConfigVersion member" does not
+     * match the actual version of the table-config-version znode we just remove all the table
+     * configurations (in the particular ZooCache instance) as was specified in the Accumulo Issue
+     * #1225. What I found is that we can usually rely on events coming to the ZCacheWatcher in
+     * order and we only need to remove one table configuration from the ZooCache. The one that just
+     * got changed, added, or deleted. This should lead to far few calls to getData on Zookeeper
+     * during runtime. This is accomplished by setting the data of the table-config-version Znode to
+     * the ZooPath of the table configuration that was modified. This zPath with be removed from the
+     * ZooCache in the ZCacheWathcer.process(). * It will be restored in the ZooCache the next time
+     * ZooCache.get(String) is called for that table configuration path. This was how ZooCached
+     * worked when we watched all table configurations and we are just emulating it now using only
+     * one watched node for all the configuration items - the "table-config-version node".
+     */
+    private synchronized void setWatcherForTableConfigVersion(WatchedEvent event) {
+
+      if (event.getPath() == null || event.getPath().isEmpty())
+        return;
+
+      if (!tableConfigWatcherSet) {
+        Matcher znodeMatcher = ZNODE_PATTERN.matcher(event.getPath());
+        if (znodeMatcher.matches()) {
+          String pathPrefix = znodeMatcher.group(1);
+          try {
+
+            Stat versionStat =
+                getZooKeeper().exists(pathPrefix + Constants.ZTABLE_CONFIG_VERSION, watcher);
+            if (versionStat != null) {
+              lastTableConfigVersion = versionStat.getVersion();
+              tableConfigWatcherSet = true;
+              if (log.isTraceEnabled())
+                log.trace("Successfully set table_config watcher");
+            }
+
+          } catch (KeeperException ke) {
+            log.error("Could not set watcher on " + pathPrefix + Constants.ZTABLE_CONFIG_VERSION
+                + " will retry later " + ke.getMessage());
+          } catch (InterruptedException ie) {
+            log.error("Could not set watcher on " + pathPrefix + Constants.ZTABLE_CONFIG_VERSION
+                + " will retry later " + ie.getMessage());
+          }
+        }
+      }
+    }
+  }
+
+  private void refreshTableConfig(byte[] configToRefresh) {
```

Comment:
add method comment

Reply:
Done.  Just pushed the new code.

Reply:
um.... I do not see a method comment here on refreshTableConfig.....

Reply:
Added the method comment and made the others I wrote conform to the previously written comments in the class.